src/nanotron/fp8/tensor.py

```diff
@@ -0,0 +1,106 @@
+import torch
+import transformer_engine as te  # noqa
+import transformer_engine_extensions as tex
+
+from nanotron.fp8.constants import DTYPE_TO_FP8_MAX, FP8_DTYPES, INITIAL_SCALING_FACTOR
+from nanotron.fp8.dtypes import DTypes
+
+
+class FP8Tensor(torch.Tensor):
+    """FP8 Tensor."""
+
+    def __new__(cls, tensor: torch.Tensor, dtype: DTypes) -> torch.Tensor:
+        # TODO(xrsrke): there is a circular import issue
+        # between tensor.py and meta.py fix this
+        from nanotron.fp8.meta import FP8Meta
```

Comment:
try to make one file import from the other? can we move convert_torch_dtype_to_te_dtype to meta.py?

Reply:
will do in another PR 🫠